### DISCUSSION TITLE
QInput Conditional Validation, Help Please

### DISCUSSION THREAD

**Discussion by leactz** [quasarframework/quasar#9152] [General - Components / Directives / etc] [answered]
I'm looping through a series of items to create a series of Q-inputs. Upon completion, the submit action for these Q-inputs will trigger an API call containing the input values as data. I need validation on each input, except for one.

So my code will look something like this:

```
      <q-form
        @submit="apiCall()">

        <div
          v-for="(item, index) in itemarray"
          :key="index" >
          <q-input
            type="text" 
            :label="index" 
            v-model="itemarray[index]"
            :rules="[ val => !!val || 'Required']" // here is where I need help
          >
          </q-input>
        </div>
        <!-- FORM ACTION BUTTON-->
        <q-btn
          icon="check"
          type="submit"
          :label="Update Item" 
        />
      </q-form>
```

The relevant part here is looping through the items in 'itemarray' to create the inputs, and then validating each of them EXCEPT in the case of `if ( itemarray[index] === 'exeptionName' )`, because 'exceptionName' is not a required field.  How can I conditionally express this in the `:rules[]` bind? The other alternative is to hardcode each input individually, but I'm sure there must be a way to conditionally validate the input based on the index of the item used to generate the input. Has somebody a suggestion for a better way?

**Comment by metalsadman** [comment, marked as answer]
Try ternary `:rules="someCondition ? [...validators] : undefined"`.

**Reply by leactz** [reply]
Thank you so much! Works perfectly.
